src/core/ext/transport/chaotic_good/client_transport.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_EXT_TRANSPORT_CHAOTIC_GOOD_CLIENT_TRANSPORT_H
+#define GRPC_SRC_CORE_EXT_TRANSPORT_CHAOTIC_GOOD_CLIENT_TRANSPORT_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stdint.h>
+
+#include <initializer_list>  // IWYU pragma: keep
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+#include "absl/base/thread_annotations.h"
+#include "absl/status/status.h"
+#include "absl/types/variant.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+#include "src/core/ext/transport/chaotic_good/frame.h"
+#include "src/core/lib/gprpp/sync.h"
+#include "src/core/lib/promise/activity.h"
+#include "src/core/lib/promise/for_each.h"
+#include "src/core/lib/promise/mpsc.h"
+#include "src/core/lib/promise/pipe.h"
+#include "src/core/lib/promise/seq.h"
+#include "src/core/lib/slice/slice_buffer.h"
+#include "src/core/lib/transport/promise_endpoint.h"
+#include "src/core/lib/transport/transport.h"
+
+namespace grpc_core {
+namespace chaotic_good {
+
+class ClientTransport {
+ public:
+  ClientTransport(std::unique_ptr<PromiseEndpoint> control_endpoint,
+                  std::unique_ptr<PromiseEndpoint> data_endpoint,
+                  std::shared_ptr<grpc_event_engine::experimental::EventEngine>
+                      event_engine);
+  ~ClientTransport() {
+    if (writer_ != nullptr) {
+      writer_.reset();
+    }
+  }
+  auto AddStream(CallArgs call_args) {
+    // At this point, the connection is set up.
+    // Start sending data frames.
+    uint64_t stream_id;
+    {
+      MutexLock lock(&mu_);
+      stream_id = next_stream_id_++;
+    }
+    bool initial_frame = true;
+    return Seq(
+        // Continuously send data frame with client to server messages.
+        ForEach(std::move(*call_args.client_to_server_messages),
+                [stream_id, &initial_frame,
```

Comment:
initial_frame = false here -- and remove the variable on the stack

Reply:
Done, thanks for catching this.